src/lib/form-field/form-field.ts

```diff
@@ -218,9 +227,13 @@ export class MatFormField extends _MatFormFieldMixinBase
       public _elementRef: ElementRef,
       private _changeDetectorRef: ChangeDetectorRef,
       @Optional() @Inject(MAT_LABEL_GLOBAL_OPTIONS) labelOptions: LabelOptions,
-      @Optional() private _dir: Directionality) {
+      @Optional() private _dir: Directionality,
+      @Optional() @Inject(MAT_FORM_FIELD_DEFAULT_OPTIONS) defaultOptions:
+          MatFormFieldDefaultOptions) {
     super(_elementRef);
 
+    this.appearance = defaultOptions && defaultOptions.appearance || 'legacy';
```

Comment:
Ternary?
```ts
this.appearance = defaultOptions ? defaultOptions.appearance : 'legacy';
```

Reply:
the appearance property on defaultOptions is optional, so I actually need to do it this way

Reply:
Ah, I think it would be better then to use an `if` statement since it's easy to misunderstand if anyone ever changes it (I misinterpreted it).